packages/igx-templates/igx-ts/custom-templates/crm-grid/files/src/app/__path__/__filePrefix__.component.ts

```diff
@@ -34,169 +36,205 @@ import { data } from './data';
-    this.overlaySettings.positionStrategy.settings.target = this.hidingButton.nativeElement;
-    this.toggleRefHiding.toggle(this.overlaySettings);
-  }
-
-  public togglePinning() {
-    this.overlaySettings.positionStrategy.settings.target = this.pinningButton.nativeElement;
-    this.toggleRefPinning.toggle(this.overlaySettings);
-  }
-
-  public ngAfterViewInit() {
-    this.grid1.summariesHeight = 60;
-    this.grid1.reflow();
-    this.cols = this.grid1.columnList;
-    this.hiddenColsLength = this.cols.filter((col) => col.hidden).length;
-    this.pinnedColsLength = this.cols.filter((col) => col.pinned).length;
-    this.crd.detectChanges();
-  }
-
-  public toggleVisibility(col: IgxColumnComponent) {
-    if (col.hidden) {
-      this.hiddenColsLength--;
-    } else {
-      this.hiddenColsLength++;
-    }
-    col.hidden = !col.hidden;
-  }
-
-  public togglePin(col: IgxColumnComponent, evt) {
-    if (col.pinned) {
-      this.grid1.unpinColumn(col.field);
-      this.pinnedColsLength--;
-    } else {
-      if (this.grid1.pinColumn(col.field)) {
-        this.pinnedColsLength++;
-      } else {
-        // if pinning fails uncheck the checkbox
-        evt.checkbox.checked = false;
-      }
-    }
-  }
-
-  public exportData() {
-    this.excelExporterService.exportData(this.localData, new IgxExcelExporterOptions('Report'));
-  }
-
-  public formatDate(val: Date) {
-    return new Intl.DateTimeFormat('en-US').format(val);
-  }
-
-  public searchKeyDown(ev) {
-    if (ev.key === 'Enter' || ev.key === 'ArrowDown' || ev.key === 'ArrowRight') {
-      ev.preventDefault();
-      this.grid1.findNext(this.searchText, this.caseSensitive);
-    } else if (ev.key === 'ArrowUp' || ev.key === 'ArrowLeft') {
-      ev.preventDefault();
-      this.grid1.findPrev(this.searchText, this.caseSensitive);
-    }
-  }
-
-  public updateSearch() {
-    this.caseSensitive = !this.caseSensitive;
-    this.grid1.findNext(this.searchText, this.caseSensitive);
-  }
-
-  public clearSearch() {
-    this.searchText = '';
-    this.grid1.clearSearch();
-  }
-
-  public formatValue(val: any): string {
-    return val.toLocaleString('en-us', { maximumFractionDigits: 2 });
-  }
+	@ViewChild("grid1", { read: IgxGridComponent, static: true })
+    public grid1: IgxGridComponent;
+
+    @ViewChild("toggleRefHiding", { static: false }) public toggleRefHiding: IgxToggleDirective;
+    @ViewChild("toggleRefPinning", { static: false }) public toggleRefPinning: IgxToggleDirective;
+
+    @ViewChild("hidingButton", { static: false }) public hidingButton: ElementRef;
+    @ViewChild("pinningButton", { static: false }) public pinningButton: ElementRef;
+
+    public localData: any[];
+    public dealsSummary = DealsSummary;
+    public earliestSummary = EarliestSummary;
+    public soonSummary = SoonSummary;
+
+    public cols: QueryList<IgxColumnComponent>;
+    public hiddenColsLength: number;
+    public pinnedColsLength: number;
+
+    public searchText: string = "";
```

Comment:
Remove implicit typing so lint doesn't have a tanrum
```suggestion
    public searchText:= '';
```